java/core/src/main/java/sleeper/core/util/PollWithRetries.java

```diff
@@ -37,25 +41,56 @@ private PollWithRetries(long pollIntervalMillis, int maxPolls) {
         this.maxPolls = maxPolls;
     }
 
+    /**
+     * Creates an instance of this class.
+     *
+     * @param  pollIntervalMillis the poll interval in milliseconds
+     * @param  maxPolls           the maximum amount of polls
+     * @return                    an instance of {@link PollWithRetries}
+     */
     public static PollWithRetries intervalAndMaxPolls(long pollIntervalMillis, int maxPolls) {
         return new PollWithRetries(pollIntervalMillis, maxPolls);
     }
 
+    /**
+     * Creates an instance of this class.
+     *
+     * @param  pollInterval the poll interval
+     * @param  timeout      the timeout used to calculate the maximum amount of polls
+     * @return              an instance of {@link PollWithRetries}
+     */
     public static PollWithRetries intervalAndPollingTimeout(Duration pollInterval, Duration timeout) {
         long pollIntervalMillis = pollInterval.toMillis();
         long timeoutMillis = timeout.toMillis();
         return intervalAndMaxPolls(pollIntervalMillis,
                 (int) LongMath.divide(timeoutMillis, pollIntervalMillis, RoundingMode.CEILING));
     }
 
+    /**
+     * Creates an instance of this class which will only poll once and never retry.
+     *
+     * @return an instance of {@link PollWithRetries}
+     */
     public static PollWithRetries noRetries() {
         return new PollWithRetries(0, 1);
     }
 
+    /**
+     * Creates an instance of this class which retry a provided number of times with no poll interval.
```

Comment:
The "with no poll interval" might be clearer as "without waiting in between polls"?